src/test/java/com/snowflake/kafka/connector/streaming/iceberg/IcebergIngestionSchemaEvolutionIT.java

```diff
@@ -107,35 +193,294 @@ private void assertRecordsInTable() {
+    waitForOffset(1);
+    List<DescribeTableRow> columns = describeTable(tableName);
+    // verify number of columns, datatype and column name
+    assertEquals(2, columns.size());
+    assertEquals("OBJECT", columns.get(1).getColumn());
+    assertEquals("OBJECT(test_string VARCHAR(16777216))", columns.get(1).getType());
+
+    // evolution
+    insertWithRetry(objectWithNestedObject, 1, withSchema);
+    waitForOffset(2);
+    columns = describeTable(tableName);
+    // verify number of columns, datatype and column name
+    assertEquals(2, columns.size());
+    assertEquals("OBJECT", columns.get(1).getColumn());
+    assertEquals(
+        "OBJECT(test_string VARCHAR(16777216), nested_object OBJECT(test_string"
+            + " VARCHAR(16777216)))",
+        columns.get(1).getType());
+
+    // evolution
+    insertWithRetry(twoObjects, 2, withSchema);
+    waitForOffset(3);
+    columns = describeTable(tableName);
+
+    assertEquals(3, columns.size());
+    // 1st column
+    assertEquals("OBJECT", columns.get(1).getColumn());
+    assertEquals(
+        "OBJECT(test_string VARCHAR(16777216), nested_object OBJECT(test_string"
+            + " VARCHAR(16777216)))",
+        columns.get(1).getType());
+    // 2nd column
+    assertEquals("OBJECT_WITH_NESTED_OBJECTS", columns.get(2).getColumn());
+    assertEquals(
+        "OBJECT(nestedStruct OBJECT(description VARCHAR(16777216)))", columns.get(2).getType());
+
+    // evolution
+    insertWithRetry(twoObjectsExtendedWithMapAndArray, 3, withSchema);
+    waitForOffset(4);
+    columns = describeTable(tableName);
+
+    assertEquals(3, columns.size());
+    // 1st column
+    assertEquals("OBJECT", columns.get(1).getColumn());
+    if (withSchema) {
+      // MAP is not supported without schema, execute this assertion only when there is a schema,
+      assertEquals(
+          "OBJECT(test_string VARCHAR(16777216), nested_object OBJECT(test_string"
+              + " VARCHAR(16777216)), Test_Map MAP(VARCHAR(16777216), OBJECT(test_string"
+              + " VARCHAR(16777216))))",
+          columns.get(1).getType());
+    }
+    // 2nd column
+    assertEquals("OBJECT_WITH_NESTED_OBJECTS", columns.get(2).getColumn());
+    assertEquals(
+        "OBJECT(nestedStruct OBJECT(description VARCHAR(16777216), test_array ARRAY(FLOAT)))",
+        columns.get(2).getType());
+  }
+
+  private static Stream<Arguments> testEvolutionOfComplexTypes_dataSource() {
+    return Stream.of(
+        Arguments.of(
+            objectVarcharWithSchema(),
+            objectWithNestedObjectWithSchema(),
+            twoObjectsWithSchema(),
+            twoObjectsExtendedWithMapAndArrayWithSchema(),
+            true),
+        Arguments.of(
+            objectVarcharPayload,
+            objectWithNestedObjectPayload(),
+            twoObjectsWithSchemaPayload(),
+            twoObjectsExtendedWithMapAndArrayPayload(),
             false));
   }
+
+  private static final String RECORD_METADATA_TYPE =
```

Comment:
Let's move it at the beginning of the class.

Reply:
Done